### DISCUSSION TITLE
Warning: Prop className did not match. Tailwind + Nextjs

### DISCUSSION THREAD

**Discussion by rodolphoasb** [tailwindlabs/tailwindcss#2015] [Help]
Hey, what's up guys?

I'm usign tailwind with Nextjs and I'm having this erros: Warning: Prop className did not match.

For better clarification of what i'm trying to do:

My authentication and persistence state process is based on cookies, since with NextJS SSR I can't acess local storage.

I have two types of navabar, one for logged out users and another for logged in user. So if the cookies are set I consider thar user Logged In and show the corresponding navbar, as follows:

`{cookies.Logged ? <NavbarLoggedIn /> : <NavbarLoggedOut />}`

Based on this Next issue _https://github.com/vercel/next.js/issues/7322_ I guess the problem is this:

The warning I'm seeing is a React hydration warning. The warning shows when the pre-rendered (SSR or SSG) response differs from the client-side React tree. In this case the className generated during pre-rendering is different from the one client-side.

My question is: Has anyone got this error? Any tips about how can I configure Tailwind with babel for working better on SSR?

**Comment by PeCrio** [comment]
You can resolve the issue by handling your `conditional rendering` with a `state` value eg. `isLoggedIn` state using the `useState()` hook, the default value can be `false`, you can use the `useEffect()` hook, to `setIsLoggedIn(cookies.Logged)` and add the `cookies.Logged` value to the dependency array of the `useEffect()` hook.

```
    const {Logged} = cookies
    const [isLoggedIn, setIsLoggedIn] = useState(false)
    useEffect(() => {
        setIsLoggedIn(Logged ? true : false)
    }, [Logged])
    
    return (
        {isLoggedIn ? <NavbarLoggedIn /> : <NavbarLoggedOut />}
    )

```

I hope this helps you or anyone out there.

**Comment by axjrl** [comment]
If someone are using persisted store, in my case it`s [Zustand persist store ](https://docs.pmnd.rs/zustand/integrations/persisting-store-data), with inline classes/styles you can try to put state out of persisted to regular [store](https://docs.pmnd.rs/zustand/integrations/persisting-store-data) it helped me!

**Reply by subhadip001** [reply]
i was using zustand persist for theme storing , and this error was coming . Now after using this 

interface ThemeStore {
  isDark: boolean;
  setIsDark: (isDark: boolean) => void;
}

const useThemeStore = create<ThemeStore>((set) => ({
  isDark: JSON.parse(localStorage.getItem("isDark") || "true"),
  setIsDark: (isDark: boolean) => {
    localStorage.setItem("isDark", JSON.stringify(isDark));
    set({ isDark });
  },
}));

Its working fine
